corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_view.js

```diff
@@ -46,9 +46,53 @@ FormplayerFrontend.module("SessionNavigate.MenuList", function (MenuList, Formpl
         },
     });
 
+    var getGridAttributes = function (tile) {
+        if (!tile) {
+            return null;
+        }
+        var rowStart = tile.gridY + 1;
+        var colStart = tile.gridX + 1;
+        var rowEnd = rowStart + tile.gridHeight;
+        var colEnd = colStart + tile.gridWidth;
+
+        return "grid-area: " + rowStart + " / " + colStart + " / " +
+            rowEnd + " / " + colEnd + ";";
+    };
+
+    function addStyleString(str) {
+        var node = document.createElement('style');
+        node.innerHTML = str;
+        document.body.appendChild(node);
+    }
+
     MenuList.CaseView = Marionette.ItemView.extend({
         tagName: "tr",
-        template: "#case-view-item-template",
+        getTemplate: function () {
+            if (_.isNull(this.options.tiles)) {
+                return "#case-view-item-template";
+            } else {
+                return "#case-tile-view-item-template";
+            }
+        },
+
+        initialize: function (options) {
+            this.tiles = options.tiles;
+            this.styles = options.styles;
+            if(!_.isNull(this.tiles)) {
+                for (var i = 0; i < this.tiles.length; i++) {
+                    var tile = this.tiles[i];
```

Comment:
in general it's best practice to declare the vars at the top since javascript using variable hoisting: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting